### PR TITLE
Feat/delegation contract

### PR DESCRIPTION
**Pull request by dvilelaf**
## Proposed changes

- Implements a wveolas delegation contract package and uses it for voting power checking

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [x] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

n/a

### CODE REVIEW

**Review by jmoreira-valory — APPROVED** [review]

File: packages/valory/skills/decision_making_abci/tasks/tweet_validation_preparation.py [hunk 1]
```diff
@@ -28,11 +28,6 @@
 )
 
 
-PROPOSAL_MINIMUM_WVEOLAS_WEI = 100e3 * 1e18  # 100K wveolas to wei
-WVEOLAS_ADDRESS = "0x4039B809E0C0Ad04F6Fc880193366b251dDf4B40"
-SERVICE_SAFE_ADDRESS = "0x12b680F1Ffb678598eFC0C57BB2edCAebB762A9A"
-
-
```
Comment (dvilelaf) [inline packages/valory/skills/decision_making_abci/tasks/tweet_validation_preparation.py:35]: Unused

File: run_service.sh [hunk 2]
```diff
@@ -17,8 +13,11 @@ autonomy fetch --local --service valory/impact_evaluator && cd impact_evaluator
 autonomy init --reset --author valory --remote --ipfs --ipfs-node "/dns/registry.autonolas.tech/tcp/443/https"
 autonomy build-image
 
+# Copy .env file
+cp $REPO_PATH/.env .
+
 # Copy the keys and build the deployment
-cp $KEY_DIR/keys.json .
+cp $REPO_PATH/keys.json .
 autonomy deploy build -ltm
```
Comment (dvilelaf) [inline run_service.sh:21]: I recommend anyone using this old approach to run services to apply these changes as well, as the export command in line 7 has some limitations with variables containing spaces. Source command also fails in some cases. Now that the framework automatically detects .env files in the build folder, this should be the recommended way. (cc @0xArdi @jmoreira-valory @Adamantios )

File: packages/valory/contracts/wveolas_delegation/contract.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+description: wveolas_delegation contract
+license: Apache-2.0
+aea_version: '>=1.0.0, <2.0.0'
+fingerprint:
+  __init__.py: bafybeidgqbp4alhzaus4ae4xlkkb4oyfion5ygkutomne3je4dn42gbtpq
+  build/DelegateContribute.json: bafybeigeu2df4b3637s7vwcbqe2vdau6cuk3rbxvkc32bg2byljidiblgm
+  contract.py: bafybeidbtq662ct7tbjbinqcft4wyr6na2y3txmcibn3rxqqodgolmkrhm
+fingerprint_ignore_patterns: []
+contracts: []
+class_name: WveOLASDelegationContract
```
Comment (DavidMinarsch) [inline packages/valory/contracts/wveolas_delegation/contract.yaml:14]: It's technically now veOLAS, can drop the w throughout

**Review by jmoreira-valory — APPROVED** [review]

File: packages/valory/skills/decision_making_abci/tasks/twitter_preparation.py [hunk 4]
```diff
@@ -369,20 +371,20 @@ def get_voting_power(self, address: str):
...
     def get_votes(
-        self, token_address, owner_address, chain_id
+        self, token_address, voter_address, chain_id
     ) -> Generator[None, None, Optional[float]]:
         """Get the given address's votes."""
         response = yield from self.behaviour.get_contract_api_response(
             performative=ContractApiMessage.Performative.GET_STATE,  # type: ignore
             contract_address=token_address,
-            contract_id=str(WveOLASContract.contract_id),
+            contract_id=str(VeOLASDelegationContract.contract_id),
```
Comment (dvilelaf) [inline packages/valory/skills/decision_making_abci/tasks/twitter_preparation.py:380]: This is the main change

**Review by Adamantios — APPROVED** [review]
Is the `wveolas_address` config now used anywhere? If not, let's open an issue to remove it.

**Comment by dvilelaf** [comment]
> Is the `wveolas_address` config now used anywhere? If not, let's open an issue to remove it.

I kept it (as I kept the contract as well). If testing goes well i'll remove the config.

**Pull request merged**